packages/flutter/test/material/icon_button_test.dart

```diff
@@ -2770,6 +2771,23 @@ void main() {
 
       expect(tester.takeException(), isNull);
     });
+
+    testWidgetsWithLeakTracking('Material3 - IconButton memory leak', (WidgetTester tester) async {
+      // This is a regression test for https://github.com/flutter/flutter/issues/130708.
+      Widget buildWidget(bool showIconButton) {
+        return showIconButton
+          ? MaterialApp(
+              theme: ThemeData(useMaterial3: true),
+              home: IconButton(
+                onPressed: () { },
+                icon: const Icon(Icons.search),
+              ),
+            )
+          : const SizedBox();
+      }
+      await tester.pumpWidget(buildWidget(true));
+      await tester.pumpWidget(buildWidget(false));
```

Comment:
nit: Sorry to add this late comment. Can we add one more comment below here to say "No exception is thrown". Because we didn't add any `expect(...)`, so just want to avoid any confusion:)